src/Components/SourcesSelect/index.js

```diff
@@ -1,35 +1,66 @@
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Alert, Select, SelectOption, Spinner } from '@patternfly/react-core';
 
 import { imageProps } from '../ProvisioningWizard/helpers';
 import { useSourcesForImage } from '../Common/Hooks/sources';
 import { useWizardContext } from '../Common/WizardContext';
 
+const LIMIT = 1;
+
 const SourcesSelect = ({ setValidation, image }) => {
   const [{ chosenSource }, setWizardContext] = useWizardContext();
   const [isOpen, setIsOpen] = React.useState(false);
   const [selected, setSelected] = React.useState(null);
+  const [offset, setOffset] = React.useState(0);
+  const [total, setTotal] = React.useState(0);
+  const [srcs, setSrcs] = useState([]);
+
   const selectObject = (id, name) => ({
     id,
     toString: () => name,
     compareTo: (other) => other.id === id,
   });
-  const { isLoading, error, sources } = useSourcesForImage(image, {
-    onSuccess: (sources) => {
-      if (chosenSource) {
-        setSelected(selectObject(chosenSource, sources.find((source) => source.id === chosenSource).name));
-        setValidation('success');
-      } else if (sources.length === 1) {
-        setSelected(selectObject(sources[0].id, sources[0].name));
-        setWizardContext((prevState) => ({
-          ...prevState,
-          chosenSource: sources[0].id,
-        }));
-        setValidation('success');
-      }
-    },
-  });
+
+  const fetchSources = (offset) => {
+    let { isLoading, error, sources, metadata } = useSourcesForImage(
+      image,
+      {
+        onSuccess: (sources) => {
+          if (chosenSource) {
+            setSelected(selectObject(chosenSource, sources.find((source) => source.id === chosenSource).name));
+            setValidation('success');
+          } else if (sources.length === 1) {
+            setSelected(selectObject(sources[0].id, sources[0].name));
+            setWizardContext((prevState) => ({
+              ...prevState,
+              chosenSource: sources[0].id,
+            }));
+            setValidation('success');
+          }
+          setSrcs((prevSrcs) => [...prevSrcs, ...sources]);
+        },
+      },
+      LIMIT,
+      offset
+    );
```

Comment:
This block needs to e called every time the offset is being changed, meaning,every time the View more button was pressed. 
But as you can see isLoading, error are dependent at that call as well. 
So I thought to have a function that is dealing with the entire fetching logic and call it every time the offset was updated. 
The browser crushes and I guess it is not the correct way, for Pubkeys we do no have that problem.

Reply:
@ezr-ondrej